compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -170,6 +170,26 @@ pub fn relate_substs_with_variances<'tcx, R: TypeRelation<'tcx>>(
     tcx.mk_substs(params)
 }
 
+#[instrument(level = "trace", skip(relation), ret)]
+fn relate_opaque_item_substs<'tcx, R: TypeRelation<'tcx>>(
```

Comment:
why do we need this instead of using `relate_substs_with_variances`? Is the `VarianceDiagInfo` that grabs the first invariant subst type causing bad diagnostics?